mermithid/processors/misc/TritiumAndEfficiencyBinner.py

```diff
@@ -0,0 +1,136 @@
+        self.fss_bins = reader.read_param(params, "fss_bins", False)
+        # If self.fss_bins is True, self.bins is ignored and overwritten
+
+        # initialize the histogram to store the corrected data
+        if self.energy_or_frequency == 'energy':
+            print(sys.getrefcount(self.corrected_data))
+            self.output_bin_variable='KE'
+        elif self.energy_or_frequency == 'frequency':
+            self.output_bin_variable='F'
+
+        if self.fss_bins == True:
+            a = self.GetEfficiencyFileContent()
+            self.bin_centers = a['frequencies']
+            self.bins = np.array(self.bin_centers) - (self.bin_centers[1]-self.bin_centers[0])/2
+            self.bins = np.append(self.bins, [self.bin_centers[-1]+(self.bin_centers[1]-self.bin_centers[0])/2])
+        else:
+            self.bin_centers = self.bins[0:-1]+0.5*(self.bins[1]-self.bins[0])
+        return True
+
+    def InternalRun(self):
+        print('namedata:',self.namedata)
+
+        N,b = np.histogram(self.data[self.namedata], self.bins)
+
+        # Efficiencies for bins
+        # If we want to use fss bins, we want False to be passed to EfficiencyAssignment
+        # because we just want to use the efficiency info directly from the file.
+        self.bin_efficiencies, self.bin_efficiency_errors = self.EfficiencyAssignment(self.bin_centers, self.bins, not self.fss_bins)
+        self.bin_efficiencies_normed = self.bin_efficiencies/np.sum(self.bin_efficiencies)
+        self.bin_efficiency_errors_normed = self.bin_efficiency_errors/np.sum(self.bin_efficiencies)
+
+        # Efficiencies for events, but with binned uncertainties (?)
+        self.event_efficiencies, self.event_efficiency_errors = self.EfficiencyAssignment(self.data[self.namedata], None, False)
+        self.event_efficiencies_normed = self.event_efficiencies/np.sum(self.event_efficiencies)
+        self.event_efficiency_errors_normed = self.event_efficiency_errors/np.sum(self.event_efficiencies)
+
+        # Put it all in a dictionary
+        temp_dictionary = {self.output_bin_variable: [], 'N': [], 'bin_efficiencies': [], 'bin_efficiency_errors': [], 'event_efficiencies': [], 'event_efficiency_errors': []}
+        temp_dictionary['N'] = N
+        temp_dictionary[self.output_bin_variable] = self.bin_centers
+        temp_dictionary['bin_efficiencies'] = self.bin_efficiencies_normed
+        temp_dictionary['bin_efficiency_errors'] = self.bin_efficiency_errors_normed
+        temp_dictionary['event_efficiencies'] = self.event_efficiencies_normed
+        temp_dictionary['event_efficiency_errors'] = self.event_efficiency_errors_normed
+        self.results = temp_dictionary
+
+        return True
+
+    def EfficiencyAssignment(self, f_bin_centers, f_bins = None, integrate_bin_width = False):
+        a = self.GetEfficiencyFileContent()
```

Comment:
This is a bit problematic to open a file into the Run part of the processor. If the file doesn't exist, the Run will fail; it is preferred that this kind of failure happens in the Configure to limit the number of failures in the Run part.
So you should move this function in the Configure

Reply:
As you open the file and reads it content, you should make sure it has all the right keys (and potentially that the values are in the format you expect, here a list I guess). Simply add a 
```python
if "frequencies" not in a.keys(): 
     logger.error("Missing "Frequencies key")
     return False
```

Reply:
Reading more carefully it seems that this file is read multiple times; maybe read it just once in the Configure and save the results in an internal variable...